src/test/java/net/adoptium/UserAgentParserTest.java

```diff
@@ -0,0 +1,116 @@
+package net.adoptium;
+
+import net.adoptopenjdk.api.v3.models.Architecture;
+import net.adoptopenjdk.api.v3.models.OperatingSystem;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class UserAgentParserTest {
+
+    @BeforeEach
+    void setUp() {
+
+    }
+
+    @Test
+    void getOsAndArchExpectedLinuxAsDefault() {
```

Comment:
Such tests are always way better readable when you follow and mark the simple "given-when-than" pattern:

```
//given
final String userAgent = "Mozilla/5.0 (X11; Linux x86_64; rv:2.1.1) Gecko/ Firefox/5.0.1";

//when
final UserSystem osArch = UserAgentParser.getOsAndArch(userAgent);

//than
assertEquals(osArch.getArch(), Architecture.x64);
assertEquals(osArch.getOs(), OperatingSystem.linux);
```